src/code/utilities/firebase-auth.js

```diff
@@ -127,15 +128,26 @@ function _userAuth(token){
   }
   return _cachedAuth;
 }
-// var connectedRef = firebase.database().ref(".info/connected");
-//   connectedRef.on("value", function(snap) {
-//     if (snap.val() === true) {
-//       console.log("Firebase connected");
-//     } else {
-//       console.log("Firebase not connected");
-//     }
-//   });
+const connectionMonitor = () => {
+  if (window.firebase) {
+    const connectedRef = firebase.database().ref(".info/connected");
+    connectedRef.on("value", function (snap) {
+      if (snap.val() === true) {
+        console.log("Firebase connected");
+        window.fbConnected = true;
+      } else {
+        console.log("Firebase not connected");
+        window.fbConnected = false;
+      }
+    });
+    return connectedRef;
+  } else return;
+};
+connectionMonitor();
```

Comment:
The connectionMonitor should immediately be triggered when firebase disconnects, right? How come this doesn't call a function that could trigger the `setConnectionState` action?

Instead the property is only being looked at on a save action. So you'll only find out you're disconnected when you make an action that doesn't get saved, instead of finding out before you make the action.

(It's only one action, true, but setting this global variable also just seemed slightly round-about. Actually, speaking of which, this would work without the `window.` but if it were just `var fbConnected` at the top, wouldn't it?)